### PR TITLE
htmlcxx: Fix requires.

### PR DESCRIPTION
**Pull request by mtl1979**
* Add missing run-time dependency for libiconv

### CODE REVIEW

File: dev-cpp/htmlcxx/htmlcxx-0.87.recipe [hunk 1]
```diff
@@ -47,6 +48,7 @@ PROVIDES_tools="
 REQUIRES_tools="
 	htmlcxx$secondaryArchSuffix == $portVersion
 	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
```
[the remark this answers is not in the document]
Comment (Begasus) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:51]: Think it should be patched: `checking for iconv_open in -liconv... no`

[the remark this answers is not in the document]
Comment (Begasus) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:51]: This fixes linking against libiconv (only for tools needed):
```diff
diff --git a/configure.ac b/configure.ac
index 302d74d..3be4864 100644
--- a/configure.ac
+++ b/configure.ac
@@ -19,7 +19,7 @@ AC_PROG_LIBTOOL
 # Checks for libraries.
 # Replace `main' with a function in -lfl:
 AC_CHECK_LIB(fl, main)
-AC_CHECK_LIB(iconv, iconv_open)
+AC_CHECK_LIB(iconv, libiconv_open)
 
 
 # Checks for header files.
```

File: dev-cpp/htmlcxx/htmlcxx-0.87.recipe [hunk 2]
```diff
@@ -38,6 +38,7 @@ PROVIDES="
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
```
Comment (Begasus) [inline dev-cpp/htmlcxx/htmlcxx-0.87.recipe:41]: Not needed here.

File: dev-cpp/htmlcxx/htmlcxx-0.87.recipe (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtl1979) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:51]: @OscarL It does link to libiconv both directly and indirectly...

[the remark this answers is not in the document]
Comment (mtl1979) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:51]: @Begasus: It does check for both iconv_open and libiconv_open to see which one is found... I didn't delete the "old" symbol name as all tests I see check for both... It makes upstreaming the patch easier.

[the remark this answers is not in the document]
Comment (Begasus) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:51]: Should look at the patchset first before commenting. :)

File: dev-cpp/htmlcxx/htmlcxx-0.87.recipe (hunk 2 above)
Comment (mtl1979) [reply dev-cpp/htmlcxx/htmlcxx-0.87.recipe:41]: Without this line, it doesn't work correctly. It complains about same missing symbol as with all recipes that depend on this recipe.

**Comment by mtl1979** [comment]
I switched the build to use `bison` in normal mode instead of yacc compatibility mode, this fixed several warnings.

**Pull request merged**